src/hotspot/share/cds/heapShared.cpp

```diff
@@ -336,6 +336,18 @@ bool HeapShared::archive_object(oop obj, oop referrer, KlassSubGraphInfo* subgra
         if (mirror_k != nullptr) {
           AOTArtifactFinder::add_cached_class(mirror_k);
         }
+      } else if (java_lang_invoke_ResolvedMethodName::is_instance(obj)) {
+        Method* m = java_lang_invoke_ResolvedMethodName::vmtarget(obj);
+        if (m != nullptr) {
+          InstanceKlass* method_holder = m->method_holder();
```

Comment:
Should you check first that the holder is already initialized before adding it to the list of AOT-inited classes?